apple/Sources/FerrostarMapLibreUI/Views/NavigationMapView.swift

```diff
@@ -22,17 +22,55 @@ public struct NavigationMapView: View {
     private var navigationState: NavigationState?
 
     @State private var locationManager = StaticLocationManager(initialLocation: CLLocation())
+
+    // MARK: Camera Settings
+
     @Binding private var camera: MapViewCamera
 
+    /// The snapped camera zoom. This is used to override the camera zoom whenever snapping is active.
+    @Binding private var snappedZoom: Double
+
+    /// Whether to snap the camera on the next navigation status update. When this is false,
+    /// the user can browse the map freely.
+    @Binding private var useSnappedCamera: Bool
```

Comment:
A few comments on this section.

1. Visibility: Unlike `State`, bindings are not private. The whole point is to be able to set them externally, so they can be public, even though the most common method of doing so is via a constructor parameter. (Aside: this also makes code cleaner in the common case since most views following this pattern can use auto-generated initializers; we only need a custom initializer because of how our builder works.)
2. No action needed on this PR, but I could see issues arising with two-way bindings and conflicting parts of the code updating the camera. Something to keep thinking about.

Reply:
Yep, it's intentionally a binding so that you can re-snap the camera (e.g. with a center on user button in a parent view/scene). In my navigation view, I'm basically using this value to hide/show my center on user button, then on tapping the button I set it back to true. Seems to work pretty well.